In [1]:
import pandas as pd
import numpy as np
import seaborn as sns


Illegal line #20
	"(44, 160, 44), (152, 223, 138), (214, 39, 40), (255, 152, 150), 
"
	in file "/Applications/anaconda/lib/python3.6/site-packages/matplotlib/mpl-data/stylelib/w209.mplstyle"
Illegal line #21
	"(148, 103, 189), (197, 176, 213), (140, 86, 75), (196, 156, 148), 
"
	in file "/Applications/anaconda/lib/python3.6/site-packages/matplotlib/mpl-data/stylelib/w209.mplstyle"
Illegal line #22
	"(227, 119, 194), (247, 182, 210), (127, 127, 127), (199, 199, 199), 
"
	in file "/Applications/anaconda/lib/python3.6/site-packages/matplotlib/mpl-data/stylelib/w209.mplstyle"
Illegal line #23
	"(188, 189, 34), (219, 219, 141), (23, 190, 207), (158, 218, 229)])
"
	in file "/Applications/anaconda/lib/python3.6/site-packages/matplotlib/mpl-data/stylelib/w209.mplstyle"
Bad val "cycler('color', [(31, 119, 180), (174, 199, 232), (255, 127, 14), (255, 187, 120)," on line #19
	"axes.prop_cycle: cycler('color', [(31, 119, 180), (174, 199, 232), (255, 127, 14), (255, 187, 120), 
"
	in file "/Applica

In [3]:
direc = '/Users/keenanszulik/dev/src/skin-cancer-mnist-ham10000/'
meta = pd.read_csv(direc + 'HAM10000_metadata.csv')

In [4]:
meta.head()

lesion_id      image_id   dx dx_type   age   sex localization
0  HAM_0000118  ISIC_0027419  bkl   histo  80.0  male        scalp
1  HAM_0000118  ISIC_0025030  bkl   histo  80.0  male        scalp
2  HAM_0002730  ISIC_0026769  bkl   histo  80.0  male        scalp
3  HAM_0002730  ISIC_0025661  bkl   histo  80.0  male        scalp
4  HAM_0001466  ISIC_0031633  bkl   histo  75.0  male          ear

In [6]:
meta.shape

(10015, 7)

In [8]:
for c in eng.columns:
    print(c)
    print(eng[c].unique())

lesion_id
['HAM_0000118' 'HAM_0002730' 'HAM_0001466' ... 'HAM_0005579' 'HAM_0001576'
 'HAM_0002867']
image_id
['ISIC_0027419' 'ISIC_0025030' 'ISIC_0026769' ... 'ISIC_0033536'
 'ISIC_0032854' 'ISIC_0032258']
dx
['bkl' 'nv' 'df' 'mel' 'vasc' 'bcc' 'akiec']
dx_type
['histo' 'consensus' 'confocal' 'follow_up']
age
[80. 75. 60. 70. 55. 85. 65. 40. 50. 45. 35.  0. 30. nan  5. 25. 20. 10.
 15.]
sex
['male' 'female' 'unknown']
localization
['scalp' 'ear' 'face' 'back' 'trunk' 'chest' 'upper extremity' 'abdomen'
 'unknown' 'lower extremity' 'genital' 'neck' 'hand' 'foot' 'acral']


In [16]:
str_cols = ['dx','dx_type','sex','localization']
for c in str_cols:
    print("\n")
    print(c)
    print(meta.groupby(c).count())



dx
       lesion_id  image_id  dx_type   age   sex  localization
dx                                                           
akiec        327       327      327   327   327           327
bcc          514       514      514   514   514           514
bkl         1099      1099     1099  1089  1099          1099
df           115       115      115   115   115           115
mel         1113      1113     1113  1111  1113          1113
nv          6705      6705     6705  6660  6705          6705
vasc         142       142      142   142   142           142


dx_type
           lesion_id  image_id    dx   age   sex  localization
dx_type                                                       
confocal          69        69    69    69    69            69
consensus        902       902   902   853   902           902
follow_up       3704      3704  3704  3704  3704          3704
histo           5340      5340  5340  5332  5340          5340


sex
         lesion_id  image_id    dx  dx_type

In [159]:
l_eight = pd.read_csv(direc + 'hmnist_8_8_l.csv')

In [160]:
l_eight.head()

pixel0000  pixel0001  pixel0002  pixel0003  pixel0004  pixel0005  \
0        172        182        191        183        180        181   
1         98        149        170        193        183        162   
2        165        164        179        172        152        163   
3        109        159        167        166        163        159   
4        173        202        210        194        208        248   

   pixel0006  pixel0007  pixel0008  pixel0009  ...    pixel0055  pixel0056  \
0        165        164        173        192  ...          159        171   
1        164        100        137        175  ...          135         83   
2        169        151        168        174  ...          169        152   
3        155         96        141        168  ...          101         79   
4        243        205        180        142  ...          230        160   

   pixel0057  pixel0058  pixel0059  pixel0060  pixel0061  pixel0062  \
0        181        201        192        184        183        171   
1        159        186        185        192        181        143   
2        171        185        189        193        176        168   
3        146        170        167        158        154        133   
4        199        206        208        209        205        200   

   pixel0063  label  
0        157      2  
1         58      2  
2        151      2  
3         45      2  
4        186      2  

[5 rows x 65 columns]

In [161]:
l_eight.shape

(10015, 65)

In [162]:
l_eight_arr = l_eight[l_eight.columns[:64]].values

In [163]:
l_eight_arr.shape

(10015, 64)

In [164]:
l_eight_arr = np.reshape(l_eight_arr, (-1,8))

In [165]:
l_eight_arr[0:8]

array([[172, 182, 191, 183, 180, 181, 165, 164],
       [173, 192, 201, 172, 176, 188, 179, 167],
       [178, 193, 185, 184, 172, 157, 196, 167],
       [176, 184, 154, 176, 145, 168, 200, 172],
       [181, 177, 186, 170, 166, 191, 175, 167],
       [181, 183, 207, 193, 163, 170, 170, 163],
       [176, 185, 209, 196, 165, 182, 171, 159],
       [171, 181, 201, 192, 184, 183, 171, 157]])

In [166]:
rgb_eight = pd.read_csv(direc + 'hmnist_8_8_RGB.csv')

In [168]:
rgb_eight.head()

pixel0000  pixel0001  pixel0002  pixel0003  pixel0004  pixel0005  \
0        199        156        188        210        165        198   
1        115         87        115        180        133        158   
2        203        146        164        209        144        153   
3        135         96        110        198        141        156   
4        204        156        187        232        185        216   

   pixel0006  pixel0007  pixel0008  pixel0009  ...    pixel0183  pixel0184  \
0        216        176        203        215  ...          214        167   
1        200        153        187        215  ...          207        167   
2        217        160        181        215  ...          211        160   
3        208        148        158        210  ...          179        141   
4        236        196        220        217  ...          236        189   

   pixel0185  pixel0186  pixel0187  pixel0188  pixel0189  pixel0190  \
0        191        199        157        175        184        143   
1        188        177        128        138         74         51   
2        175        199        152        172        177        138   
3        157        153        122        138         58         39   
4        213        233        182        209        207        175   

   pixel0191  label  
0        167      2  
1         60      2  
2        158      2  
3         46      2  
4        188      2  

[5 rows x 193 columns]

In [170]:
rgb_eight.shape

(10015, 193)

## RBG 8_8

This data is slightly different than the previous, with 192 different pixels (as opposed to the 64 identified in `L 8_8`). These images are larger than the previous data we looked at.